### PR TITLE
GPG-792 employers opt out of reporting pay quarters

### PR DESCRIPTION
**Pull request by malinaslevoaca1**
See ticket [GPG-792](https://technologyprogramme.atlassian.net/browse/GPG-792)
Employers can opt out of reporting pay quarters for years with furlough scheme.

Added tests but also tested locally:
* The behavior isn't changed for other reporting years - the pay quarters are required and must add up to 100.
* If not opted out, pay quarters are required and must add up to 100.
If opted out:
* pay quarters cannot be submitted
* pay quarters are marked as "Not applicable" on the check data page
* pay quarters are not displayed on the report page

image
image
image

### CODE REVIEW

File: GenderPayGap.WebUI/Views/AdminOrganisationReturn/ViewReturnDetails.cshtml [hunk 1]
```diff
@@ -307,9 +307,9 @@
                                           (thisVersion.FemaleUpperPayBand != previousVersion.FemaleUpperPayBand));
 
                         <td class="govuk-table__cell @(isChanged ? "value-changed-from-previous-return" : null)">
-                            <span style="white-space: nowrap;">@(thisVersion.MaleUpperPayBand.ToString("0"))% male</span>
+                            <span style="white-space: nowrap;">@(thisVersion.MaleUpperPayBand?.ToString("0"))% male</span>
```
Comment (Lucas-Fernandes-Nogueira) [inline GenderPayGap.WebUI/Views/AdminOrganisationReturn/ViewReturnDetails.cshtml:310]: Should we not display the table at all if there's no values?

**Review by Lucas-Fernandes-Nogueira — APPROVED** [review]
Looks great Malina! Really nice work! I just left a small question but consider this approved 😄.

File: GenderPayGap.WebUI/Views/AdminOrganisationReturn/ViewReturnDetails.cshtml (hunk 1 above)
Comment (malinaslevoaca1) [reply GenderPayGap.WebUI/Views/AdminOrganisationReturn/ViewReturnDetails.cshtml:310]: So this table is used for displaying all the versions of a report and it includes all the figures (not only the pay quarters but also the bonuses and the mean, median figures).

Comment (Lucas-Fernandes-Nogueira) [reply GenderPayGap.WebUI/Views/AdminOrganisationReturn/ViewReturnDetails.cshtml:310]: Will this table sometimes show an empty cell with just the word male in it though?

Comment (malinaslevoaca1) [reply GenderPayGap.WebUI/Views/AdminOrganisationReturn/ViewReturnDetails.cshtml:310]: Yes, it's the same for the bonus figures that are already optional - they have "%" displayed. We can definitely change that but I don't think it really matters as it's an admin table.

We might want to add a description (eg "Employer opted out of reporting pay quarters") but I will check that with GEO and we can make the changes as part of the other ticket when we update the other copies as well.

Thanks for flagging this!

Comment (Lucas-Fernandes-Nogueira) [reply GenderPayGap.WebUI/Views/AdminOrganisationReturn/ViewReturnDetails.cshtml:310]: Great, that makes sense. Thanks Malina 😄.

**Pull request merged**